ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -828,10 +828,23 @@ void writeIndexSegmentToFile(Offset safeEndPoint) throws FileNotFoundException,
    * @throws StoreException if there are problems with the index
    */
   void seal() throws StoreException {
-    sealed.set(true);
-    map();
-    // we should be fine reading bloom filter here without synchronization as the index is read only
-    persistBloomFilter();
+    rwLock.writeLock().lock();
+    try {
+      sealed.set(true);
+      map();
+      // we should be fine reading bloom filter here without synchronization as the index is read only
```

Comment:
do we want to move this outside of the rwlock if its okay to read w/o synchronization? then we wouldn't need the index != null check in the catch block